### ISSUE TITLE
Figma "Preview" mode doesn't work

### ISSUE THREAD

**Issue by felschr** [mullvad/mullvad-browser#135]
Using the "Preview" mode in Figma doesn't work.
It renders a loading bar that slowly progresses but never finishes loading.

Here is an example Figma project:
https://www.figma.com/file/GBCGqs11KNeUyPksQzEMw0hr/Wireframing/duplicate?source=website

The link above opens the project in the "Present" mode which seems to work fine.

Using the play button in the top right one can open the "Preview" mode. Here is a direct link to it:
https://www.figma.com/proto/GBCGqs11KNeUyPksQzEMw0hr/Wireframing?type=design&node-id=0-53&scaling=min-zoom&page-id=0%3A1

**Comment by Thorin-Oakenpants** [comment]
some notes
- On FF I enabled RFP and with or without a canvas exception, the preview loaded. Note, the only extension I had in nightly was uBO. So it doesn't "seem" to be RFP
- In MB I can replicate the issue
  - I disabled NoScript, RFP and uBO and it works

over to rui

**Comment by Thorin-Oakenpants** [comment]
using https://www.figma.com/proto/GBCGqs11KNeUyPksQzEMw0hr/Wireframing?type=design&node-id=0-53&scaling=min-zoom&page-id=0%3A1

RFP on (fails)

![RFPon](https://github.com/mullvad/mullvad-browser/assets/16656956/54ede35c-f044-4bfc-9a38-92c2231cb89b)

RFP off (works)

![RFP-off](https://github.com/mullvad/mullvad-browser/assets/16656956/63466d71-f049-41ce-9212-c12b6c9419c3)

**Comment by Thorin-Oakenpants** [comment]
So this is a little weird to me, because RFP in FF doesn't seem to cause an issue, but does in MB. And the code is minified, so we'll have to dig deeper on the uncaught in promise error

**Comment by felschr** [comment]
I just disabled RFP on Mullvad Browser and that Figma link starts working for me.
I guess that's probably because I also have Private Browsing mode disabled.

**Comment by Thorin-Oakenpants** [comment]
good point. I was using normal windows in FF. Normally the only diff (for websites) between normal and PBM is service workers (IDB was added in time for TB/MB13).

I'll test FF in PBM to see if it replicates

**Comment by Thorin-Oakenpants** [comment]
> I'll test FF in PBM to see if it replicates

The test works perfectly fine in FF with RFP + in a PB window (and ETP was strict, and uBO was enabled)

**Comment by Thorin-Oakenpants** [comment]
the culprit seems to be webgl related - e.g. if you disable it `webgl.disabled` = `true` - this is not what TB/MB sets, it was a simulation

TB/MB set this (standard mode)
```js
// tor-browser#18603: failIfMajorPerformanceCaveat is an optional attribute that
// can be used when creating a WebGL context if the browser detects that the
// performance would be low. That could be used to fingerpting users with a not
// good graphics card, so disable it (actually, defense in depth, it is already
// disabled by default from Firefox).
pref("webgl.disable-fail-if-major-performance-caveat", true);
// tor-browser#16404: disable until we investigate it further (#22333)
pref("webgl.enable-webgl2", false);
```

We can ignore the first pref, and the second, tested at false on windows, didn't affect the test. One of the errors thrown when it does fail in TB/MB was a line that included `readPixels()` - and TB do block this at standard and put it behind click-to-play on safer

you can see the difference here: https://registry.khronos.org/webgl/sdk/tests/conformance/reading/read-pixels-test.html - TB/MB just utterly fail to even enumerate the tests, FF passes 209 and fails none  for example

**Comment by Thorin-Oakenpants** [comment]
testing on safer, click to play is never triggered by NS

on the direct link the test just progresses and never finishes

on https://www.figma.com/file/GBCGqs11KNeUyPksQzEMw0hr/Wireframing/duplicate?source=website I get the following

![safer](https://github.com/mullvad/mullvad-browser/assets/16656956/c7984332-08e6-4286-9e6f-17a2fb09a2c0)

@hackademix

**Comment by ruihildt** [comment]
It turns out that figma is now straight up blocking the website from rendering and redirects to https://www.figma.com/unsupported_browser

Here's what they say they support: https://help.figma.com/hc/en-us/articles/360039827194-What-browsers-does-Figma-support
